frontend/device/devicelistener.lua

```diff
@@ -51,24 +49,38 @@ end
 function DeviceListener:onShowWarmth(value)
     local powerd = Device:getPowerDevice()
     if powerd.fl_warmth ~= nil then
-        UIManager:show(Notification:new{
-            text = T(_("Warmth set to %1."), powerd.fl_warmth),
-            timeout = 1.0,
-        })
+        if powerd.fl_warmth_max == nil then
+            UIManager:show(Notification:new{
+                text = T(_("Warmth set to %1."), math.floor(powerd.fl_warmth)),
+                timeout = 1.0,
+            })
+        else
+            UIManager:show(Notification:new{
+                text = T(_("Warmth set to %1."), math.floor(powerd.fl_warmth/100*powerd.fl_warmth_max)),
```

Comment:
I don't understand this one.

What is the intention? if fl_warmth == fl_warmth_max the result will be 0.01. Am I missing something? Sorry but I didn't sleep very well :p

Reply:
ok I added parenthesis :p. I guess. I still don't understand but maybe is my bad

Reply:
It is annoying if you have a slider (gesture) showing values from 0..100, but the warmth only changes on 0,10,20,30...100.
In this PR the message of the gesture should be changed, if the devices `fl_warmth_max - fl_warmth_min` is much smaller than the **default warmth range (0..100)**.

Internally nothing is changed. The whole frontend works with the default warmth range (0..100). It always applies that **0<=warmth<=100**

The only changes are (only on smaller scaled devices):
1.) The Notification will show values from fl_warmth_min..fl_warmth_max: e.g. with warmth = 58 and fl_warmth_max=10 then floor(58/100*10) -> 5
2.) The scale for swiping must be greater, if `fl_warmth_max - fl_warmth_min` is smaller than the original 100. I find the new `steps_fl_small_scale` very comfortable, but it is not engraved in stone.

Kobos and other warmth devices are not touched by this PR, up to now it will only affect android e-Ink devices.

Reply:
I'm not *quite* fond of showing the internal implementation detail only in *some* places like this (i.e., specifically talking about 1.) here).

I think it makes much more sense is we *always* display a [0, 100] scale to users, no matter the internal implementation.

Now, that's coming from a Kobo, where the stock UI doesn't actually show numbers, just a progress bars.
If the stock UI itself displays those numbers, that's another kettle of fish, and I'd tend to agree with you, with a caveat: make it consistent ;).

(I have no qualms about tweaking the swipe tweak, although I've never found it necessary myself on Kobo, FWIW).

Reply:
> I think it makes much more sense is we always display a [0, 100] scale to users, no matter the internal implementation.

Me too, but I understand the idea behind the change. I'm also a bit of lover of little aesthetics ultraoptimizations :p. In that case my Bq does the same as a Tolino or a Forma: it shows 0 - 100 but just changes warmth each 10 steps. I'm fine with current behaviour but a 1 - 10 message won't bother me either.

In any case I think the logic has some flaws. I will review in a moment.

Reply:
@pazos Just for understanding: I suppose that your Bq is a phone? 

@pazos and @NiLuJe: What shall we do? A scale from `0-100` or `0-wl_warmth_max`. If we have very few steps, I would prefer the small scale. But anyway, if the decission will be `0-100`, shouldn't we add a %-symbol to the number (and also to the brightness number)?

Reply:
The regular frontlight adjusts its steps, doesn't it? It has a 100 (?) on my Kobo.
![Screenshot_2020-08-05_08-44-16](https://user-images.githubusercontent.com/202757/89380496-dd857a80-d6f7-11ea-9a44-66da25dc3c4b.png)

Reply:
@frenzie: Is this the original "Frontlight dialog" on your Kobo?

Reply:
No, on my Kobo it's 0-100. This is a scale from 0-10 or so as on the emulator and presumably some older Kindle that it's mimicking. Which is my point, that it adapts.

Reply:
Yep, the frontlight widget adapts it (the same on android). Here we're talking about the little message that pops up when you perform a warmth change with a gesture. Right now is 0 - 100. With this change is 0 - 10. It is fine in devices with 10  warmth steps but devices with more warmth steps need more resolution.

Solution? I don't know. I'm fine with the current behaviour and I'm fine too adding a % symbol to it.

Reply:
> ... It is fine in devices with 10 warmth steps but devices with more warmth steps need more resolution.

The popup shows values between 0 and `fl_warmth_max`. So it adapts the resolution itself.

Reply:
Huh, I was convinced the FL widget was always showing a [0,100], my bad ;).